core/src/main/java/io/questdb/cairo/O3PartitionJob.java

```diff
@@ -52,6 +60,89 @@ public O3PartitionJob(MessageBus messageBus) {
         super(messageBus.getO3PartitionQueue(), messageBus.getO3PartitionSubSeq());
     }
 
+    public static void processParquetPartition(
+            Path pathToTable,
+            int partitionBy,
+            ReadOnlyObjList<? extends MemoryCR> oooColumns,
+            long srcOooLo,
+            long srcOooHi,
+            long partitionTimestamp,
+            long sortedTimestampsAddr,
+            TableWriter tableWriter
+    ) {
+
+        // Number of rows to insert from the O3 segment into this partition.
+        final long srcOooBatchRowSize = srcOooHi - srcOooLo + 1;
+
+        final TableRecordMetadata tableWriterMetadata = tableWriter.getMetadata();
+        try (PartitionDecoder file = new PartitionDecoder(tableWriter.getFilesFacade())) {
+            final Path path = Path.getThreadLocal(pathToTable);
+            TableUtils.setPathForPartition(path.trimTo(pathToTable.size()), partitionBy, partitionTimestamp, -1);
+            path.put(".parquet");
+            file.of(path.$());
+
+
+            final int rowGroupCount = file.getMetadata().rowGroupCount();
+            final int timestampIndex = tableWriterMetadata.getTimestampIndex();
+            final int timestampColumnType = tableWriterMetadata.getColumnType(timestampIndex);
+
+            long mergeRangeLo = srcOooLo;
+            for (int rowGroup = 1; rowGroup < rowGroupCount; rowGroup++) {
+                final long min = file.getColumnChunkMinTimestamp(rowGroup, timestampIndex);
+                final long mergeRangeHi = Vect.boundedBinarySearchIndexT(
+                        sortedTimestampsAddr,
+                        min,
+                        mergeRangeLo,
+                        srcOooHi,
+                        BinarySearch.SCAN_DOWN
+                );
+                // no data in the O3 segment for this row group
+                if (mergeRangeHi < mergeRangeLo) {
+                    continue;
+                }
+                mergeRowGroup(
+                        oooColumns,
+                        sortedTimestampsAddr,
+                        tableWriter,
+                        file,
+                        rowGroup - 1,
+                        timestampIndex,
+                        timestampColumnType,
+                        mergeRangeLo,
+                        mergeRangeHi,
+                        tableWriterMetadata,
+                        srcOooBatchRowSize,
+                        path);
+                mergeRangeLo = mergeRangeHi + 1;
+            }
+
+            if (mergeRangeLo <= srcOooHi) {
+                // merge the tail into the last row group
+                //[mergeRangeLo, srcOooHi]
+                mergeRowGroup(
```

Comment:
It would make sense to append a new row group if all the o3 data timestamps > last row group max timestamp and the last row group is already quite big.